docs/source/development/cdf_guide.rst

```diff
@@ -0,0 +1,686 @@
+
+The internal format of CDF files are described `here <https://cdaweb.gsfc.nasa.gov/pub/software/cdf/doc/cdf391/cdf39ifd.pdf>`_.
+
+Variables
+----------
+
+.. image:: ../_static/cdf_variables.png
+
+
+Variables are the data stored within a file.  They have a few key characteristics -
+
+* Name
+   * Variables are given a unique name
+* Shape
+   * The the number and the size of the dimensions.  In the picture above, "zVariable 1" is 5x5, and "zVariable 2" is 5x2.
+* Data types
+   * Describes what the data actually represents, some options include -
+      * Integers (1-8 bytes)
+      * Floats (1-8 bytes)
+      * Characters (1 byte)
+      * "Time" data
+         *  For IMAP, the main time variable we'll be using is CDF_EPOCH_TT2000, which is nanoseconds since the year 2000.  More detail can be found in below sections.
+* Records
+   * Variables can span across multiple records.  This is essentially another dimension of the data, but the idea is that each record represents 1 unique physical measurement from an instrument.  In the above diagram, both variables have 3 records shown.
+
+.. note:: For IMAP instruments, each "Record" is always a unit of time.  If a varible does not vary over time, it will only span 1 record.
+
+Attributes
+-----------
+
+Attributes are essentially key:value pairs inside the CDF, similar to a python dictionary object.  For example - Mission=IMAP, or Instrument=IDEX.
+
+Attributes come in two forms:
+
+* **Variable Attributes** - Attributes can be attached to a Variable if they only describe a particular data variable.
+* **Global Attributes** - If they are not attached to a variable, they describe the file in it's entirety.
+
+
+ISTP Compliance
+===============
+
+ISTP compliance is a standard set of attributes to use when creating CDF files.  The goal of ISTP is that others users and programs of your CDF file will know where to look for information about the data inside.  A large part of these requirements are driven by the SPDF's display tool, `CDAWeb <https://cdaweb.gsfc.nasa.gov/cdaweb/>`_.
+
+.. image:: ../_static/2d_spectrogram_1.gif
+
+
+The image above is taken from CDAWeb. Several of the attributes inside the CDF file are used to help generate these plots.
+
+To archive at the SPDF, files must be ISTP compliant.  The SPDF will reject files that do not conform to their standards.
+
+A guide to SPDF can be found here: `https://spdf.gsfc.nasa.gov/istp_guide/istp_guide.html <https://spdf.gsfc.nasa.gov/istp_guide/istp_guide.html>`_
+
+*****************************
+cdflib.xarray_to_cdf
+*****************************
+
+The Python library `cdflib <https://github.com/MAVENSDC/cdflib>`_ is a pure-python implementation of the CDF libraries.  Of particular note is the function `xarray_to_cdf <https://cdflib.readthedocs.io/en/latest/api/cdflib.xarray.xarray_to_cdf.html#cdflib.xarray.xarray_to_cdf>`_.
```

Comment:
why is xarray_to_cdf of particular note? Maybe mention something about the IMAP SDC using xarray for all internal data representations.